video/out/vulkan/context_wayland.c

```diff
@@ -90,13 +92,14 @@ static bool wayland_vk_init(struct ra_ctx *ctx)
         goto error;
     }
 
-    /* Because in Wayland clients render whenever they receive a callback from
-     * the compositor, and the fact that the compositor usually stops sending
-     * callbacks once the surface is no longer visible, using FIFO here would
-     * mean the entire player would block on acquiring swapchain images. Hence,
-     * use MAILBOX to guarantee that there'll always be a swapchain image and
-     * the player won't block waiting on those */
-    if (!ra_vk_ctx_init(ctx, vk, params, VK_PRESENT_MODE_MAILBOX_KHR))
+    /* If the Wayland compositor does not support commit-timing, fifo, and
+     * presentation time v2 protocols, the compositor will stop sending callbacks
+     * if the surface is no longer visible. This means using FIFO would block
+     * the entire vo thread which is just not good. Use MAILBOX for those
+     * compositors to avoid indefinite blocking. */
+    struct vo_wayland_state *wl = ctx->vo->wl;
+    p->use_fifo = wl->has_commit_timing && wl->has_fifo && wl->present_v2;
```

Comment:
if user sets `wayland-internal-vsync=yes` and fifo is available, this will use fifo instead of mailbox. Is that expected?

Reply:
Ah missed that. I guess we should also make sure that wayland option isn't `yes`.